src/libsyntax/feature_gate.rs

```diff
@@ -409,6 +409,12 @@ impl<'a> Context<'a> {
 
 pub fn emit_feature_err(diag: &SpanHandler, feature: &str, span: Span, explain: &str) {
     diag.span_err(span, explain);
+
+    // #23973: do not suggest `#![feature(...)]` if we are in beta/stable
```

Comment:
(In particular, the detail of *why* we are early returning requires one read beneath the match statement ... I figure a one-line comment with issue number is worthwhile in that context.  If I had done this by moving the `fileline_help` invocations into the `_ => { ... }` arm instead, then I would not have included the comment, and would have treated the code as self-explanatory in that case.  But as soon as I fiddle with `return` or `break` or other abnormal control-transfers, I try to err on the side of leaving more comments.)